app/services/transfer_stock_service.rb

```diff
@@ -0,0 +1,19 @@
+class TransferStockService
+  def initialize(product:, source_location:, destination_location:, quantity:)
+    @product = product
+    @source_location = source_location
+    @destination_location = destination_location
+    @quantity = quantity.to_i
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      @stock_1 = Stock.find_by(location: @source_location, product: @product)
```

Comment:
This code is fine but `@product.stocks.find_by(location: @source_location)` is more rails way I think :) Also shouldn't we use `find_by!` here?

Reply:
Also, I don't think we need to use instance variables for stock_1 and stock_2